fe/fe-core/src/main/java/com/starrocks/planner/PlanFragment.java

```diff
@@ -672,4 +674,25 @@ public List<OlapScanNode> collectOlapScanNodes() {
 
         return olapScanNodes;
     }
+
+    public boolean isUnionFragment() {
```

Comment:
using the recursive function is just ok? why bfs-search?

Reply:
I just move this method from `CoordinatorPreprocessor::isUnionFragment` to `PlanFragment`, and don't change the original logic.